server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1776,6 +1777,8 @@ public Store.MetadataSnapshot snapshotStoreMetadata() throws IOException {
     public Map<String, StoreFileMetadata> getSegmentMetadataMap() throws IOException {
         try (final GatedCloseable<SegmentInfos> snapshot = getSegmentInfosSnapshot()) {
             return store.getSegmentMetadataMap(snapshot.get());
+        } catch (IOException e) {
+            throw new OpenSearchCorruptionException("Error fetching local metadata");
```

Comment:
Javadoc is now incorrect. Also this feels a bit like a hack to me.

Reply:
It is, fixing it.  SegRep itself will only fail the shard if OpenSearchCorruptionException is the reason for failure.  We can just look for CorruptIndexException rather than re-wrapping.

Reply:
also what this fixes is not likely but would prefer to be defensive. A file would have to be deleted underneath us before segrep runs.  If that happens we can get in a loop of failure at this call and retrying indefinitely.